Source/UnitTests/ExecutionEngineTests/ExecutionEngineTest.cs

```diff
@@ -253,9 +253,10 @@ public async Task LoopInvariantDescriptions()
 
     Assert.AreEqual((firstName, new Running()), statusList[0]);
     Assert.AreEqual((secondName, new Queued()), statusList[1]);
-    Assert.AreEqual(firstName, statusList[2].Item1);
-    Assert.IsTrue(statusList[2].Item2 is Completed);
-    Assert.AreEqual((secondName, new Running()), statusList[3]);
+    var indexTwoAndThreeOrdered = statusList.Skip(2).Take(2).OrderByDescending(k => k.Item1 == firstName).ToList();
```

Comment:
It has 5 items, this takes the third and fourth

Reply:
And is the argument that elements 0 and 1 will always be in the right order because reaching the `Running` or `Queued` state is immediate and `firstStatuses.Subscribe` is called before `secondStatuses.Subscribe`?

Reply:
I'm not sure whether it the order of 0 and 1 is guaranteed, but I haven't seen it cause issues so far.

Reply:
Ok, I get it now why there is 4 statuses, since it's a concat of the statuses. What's the reason to concat them and not inspect them separately?
And what is the 5th status then?
And why do we care about priorities in the source code if we are sorting them anyway?